### PR TITLE
SortGff tool

### PR DESCRIPTION
**Pull request by kachulis**
Adds a tool to sort and add flush directives to a gff3 file.  Requires samtools/htsjdk#1486

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/annotation/SortGff.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                comp = f1.getStart() - f2.getStart();
+            }
+
+            return comp;
+        }
+    }
+
+    @Override
+    protected int doWork() {
+        final Gff3Codec inputCodec = new Gff3Codec(Gff3Codec.DecodeDepth.SHALLOW);
```
Comment (jessicaway) [inline src/main/java/picard/annotation/SortGff.java:120]: consider adding input and output checking before computing
`IOUtil.assertFileIsReadable(INPUT);
  IOUtil.assertFileIsWritable(OUTPUT);`

**Comment by kbergin** [comment]
Hey @kachulis - just going through open Picard PRs. Would you be able to respond to these comments to merge this PR?

**Comment by kachulis** [comment]
sure!

**Comment by kachulis** [comment]
Thanks @jessicaway, sorry about the super long lag time.  I have added the input/output checks requested, and rebased.

**Comment by gbggrant** [comment]
Hi @kachulis looks like a couple of tests are failing.  We're going to wait for you to update the tests before reviewing.

**Comment by kachulis** [comment]
@gbggrant, fixed, looks like it's passing now

File: src/main/java/picard/annotation/SortGff.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    public File SEQUENCE_DICTIONARY;
+
+    @Argument(doc = "Number of records to hold in memory before spilling to disk", optional = true)
+    public int nRecordsInMemory = 50000;
+
+    private final Log log = Log.getInstance(SortGff.class);
+
+    private final Map<String, Integer> latestStartMap = new HashMap<>();
+    private int latestStart = 0;
+    String latestChrom;
```
Comment (gbggrant) [inline src/main/java/picard/annotation/SortGff.java:100]: Can this be private?

**Review by gbggrant — APPROVED** [review]

**Comment by jessicaway** [comment]
@kachulis this is ready to merge when you get a chance

**Pull request merged**